client/me/purchases/manage-purchase/change-payment-method/index.jsx

```diff
@@ -361,6 +373,43 @@ function TosText( { translate } ) {
 	);
 }
 
+function CurrentPaymentMethodNotAvailableNotice( { purchase } ) {
+	const translate = useTranslate();
+	const moment = useLocalizedMoment();
+	const storedPaymentAgreements = useSelector( getStoredPaymentAgreements );
+
+	let noticeText = '';
+
+	if ( creditCardHasAlreadyExpired( purchase ) ) {
+		noticeText = translate( 'Your %(cardType)s ending in %(cardNumber)d expired %(cardExpiry)s.', {
```

Comment:
This is fine, but we could avoid having the `else if`/`else` as well as the `let noticeText` and the conditional return (with its implicit but maybe not obvious possible `false` response) if we changed each of these to an early return. This would make it easier to see what each condition returned without having to follow the whole function and will more importantly help keep complexity down when more conditions are inevitably added in the future. We could then make the default `return null;` which is a more semantic React component return value than `false`.

Reply:
I had it that way originally (I actually had this as a regular helper function which only returned the text, and then it was easy to do early returns), but the linter complained to me that I shouldn't use `useSelector()` and the others outside of a React component... So that's when I converted it to a component with a single return value.

I guess we could still do early returns of the `<Notice>` component itself; it's a little repetitive, but since we're only passing one non-default parameter to that component, I guess it's OK.  Yeah, probably worth doing.

Reply:
Actually, I guess we can just define the default props to `<Notice>` out in front and then avoid any duplication that way.  I went ahead and did that.